### PR TITLE
WT-11898 Rename catch2 unit tests in evergreen to mention catch2

### PR DESCRIPTION
**Pull request by ajmorton**
Small naming change so that our unit tests explicitly mention catch2 to reduce the naming collision between these tests and the Python `unit-test-bucket*`. We've decided to rename the catch2 tests as they're newer and long running metrics may be impacted by a rename of our python tests.

Once this is in we'll add a new "Catch2 Unit Tests" option in the Jira `Components` field

### CODE REVIEW

**Comment by jeremythorp** [comment]
Hi Andrew, that change LGTM. Before we merge, please could you manually launch the ‘Code statistics’ tests in your PR patch build so we can check they still work Thank, Jeremy.

**Review by jeremythorp — APPROVED** [review]
LGTM (assuming the 'Code statistics' tests pass.

**Comment by ajmorton** [comment]
Confirmed that the `Code statistics` tests are all passing. Merging PR

**Comment by ajmorton** [comment]
evergreen merge

**Comment by ajmorton** [comment]
We've had a failure in the commit queue, but this is caused by WT-11669 and not from this PR.
Merging manually.

**Pull request merged**
